lib/v8.js

```diff
@@ -399,19 +407,63 @@ function deserialize(buffer) {
 
 class GCProfiler {
   #profiler = null;
+  #options = {
+    mode: GC_PROFILER_MODE.ALL,
+  };
+  constructor(options) {
+    if (options) {
+      validateObject(options);
+      if (options.mode !== undefined) {
+        validateUint32(options.mode, 'options.mode');
+        validateOneOf(options.mode,
+                      'options.mode',
+                      [
+                        GC_PROFILER_MODE.TIME,
+                        GC_PROFILER_MODE.ALL,
+                      ]);
+        this.#options.mode = options.mode;
+      }
+    }
+  }
+
+  static get GC_PROFILER_MODE() {
+    return GC_PROFILER_MODE;
+  }
+
+  static get GC_PROFILER_FORMAT() {
+    return GC_PROFILER_FORMAT;
+  }
 
   start() {
     if (!this.#profiler) {
-      this.#profiler = new binding.GCProfiler();
+      this.#profiler = new binding.GCProfiler(this.#options.mode);
       this.#profiler.start();
     }
   }
 
-  stop() {
+  getTotalGCTime() {
     if (this.#profiler) {
-      const data = this.#profiler.stop();
-      this.#profiler = null;
-      return JSONParse(data);
+      return this.#profiler.getTotalGCTime();
+    }
+  }
+  // Backward compatibility
+  stop(format = GC_PROFILER_FORMAT.OBJECT) {
+    if (format !== undefined) {
+      validateUint32(format, 'format');
+      validateOneOf(format,
+                    'format',
+                    [
+                      GC_PROFILER_FORMAT.NONE,
+                      GC_PROFILER_FORMAT.OBJECT,
+                      GC_PROFILER_FORMAT.STRING,
+                    ]);
+    }
+    if (this.#profiler) {
+      try {
+        return this.#profiler.stop(format);
```

Comment:
I think you can just get the result as a string from the C++ binding unconditionally and parse it from JS land?